go.mod

```diff
@@ -1,119 +1,101 @@
 module sigs.k8s.io/cluster-api-provider-azure
 
-go 1.22.7
+go 1.23.2
 
-toolchain go1.22.12
+toolchain go1.24.1
```

Comment:
CAPZ's go version updates are driven by CAPI minor updates. 
Currently CAPZ main is at CAPI v1.9.x. Once we merge CAPI v1.10.0, we can review this PR (Reference: https://github.com/kubernetes-sigs/cluster-api/blob/release-1.10/go.mod#L3)
/hold

Reply:
/cc @mboersma @willie-yao @Jont828